src/par/FixColumnPartitions.c

```diff
@@ -99,18 +114,71 @@ int FixColumnPartitions_UpDownFaces(Mesh_ptr mesh, MRegion_ptr mr, MFace_ptr* up
     MSTK_Report("FixColumnPartitions","Mesh is not quite columnar, up and down faces do not align.",MSTK_FATAL);
   }
 
-  if (upcen[2] < dncen[2] - 1.e-12) {
-    rf = *up;
-    *up = *dn;
-    *dn = rf;
-    ret = 1;
-  } else if (upcen[2] > dncen[2] + 1.e-12) {
-    ret = 1;
-  } else {
-    /* find the first region upward that is proper */
-    MSTK_Report("FixColumnPartitions","Partitioning with degenerate cells is not yet supported.",MSTK_FATAL);
-    ret = 0;
+  if (up_unknown == 1) {
```

Comment:
Note that you should never use the difference between face centroids and cell centroids to define a up or down -- this has gotten us in trouble several times and been removed from several times.  Face centroids of the "up" face can still have a z coordinate that is less then cell centroid z coordinate.